lib/fluent/plugin/formatter_syslog_rfc5424.rb

```diff
@@ -23,18 +23,20 @@ def configure(conf)
         @log_field_array = @log_field.split(".")
       end
 
-      def format(tag, time, record)
+      def format(tag, time, record, token = nil)
         log.debug("Record")
         log.debug(record.map { |k, v| "#{k}=#{v}" }.join('&'))
 
+        structured_data = token.nil? nil: RFC5424::StructuredData.new(sd_id: token).to_s
+
         msg = RFC5424::Formatter.format(
           log: record.dig(*@log_field_array) || "-",
           timestamp: time,
           hostname: record.dig(*@hostname_field_array) || "-",
           app_name: record.dig(*@app_name_field_array) || "-",
           proc_id: record.dig(*@proc_id_field_array) || "-",
           msg_id: record.dig(*@message_id_field_array) || "-",
-          sd: record.dig(*@structured_data_field_array) || "-"
+          sd: structured_data || record.dig(*@structured_data_field_array) || "-"
```

Comment:
If the token is present, then it won't gather the structured data field array, I assume we want to dig first, and if present, add the sd_id, otherwise send -

Reply:
True, Not sure if we are using it anywhere else. But yes ill add a check so it appends to existing data if any. Wanted to confirm approach was correct :) Thanks for the quick response !